test_app/lib/screens/participant_screen/spatial_position_dialog_content.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:dolbyio_comms_sdk_flutter/dolbyio_comms_sdk_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import '../../widgets/dialogs.dart';
+
+class SpatialPosiotionDialogContent extends StatefulWidget {
+  final Participant participant;
+  final BuildContext spatialPositiontDialogContext;
+  final BuildContext resultDialogContext;
+  const SpatialPosiotionDialogContent(
+      {Key? key,
+      required this.participant,
+      required this.spatialPositiontDialogContext,
+      required this.resultDialogContext})
+      : super(key: key);
+
+  @override
+  State<SpatialPosiotionDialogContent> createState() =>
+      SpatialPosiotionDialogContentState();
+}
+
+class SpatialPosiotionDialogContentState
+    extends State<SpatialPosiotionDialogContent> {
+  final formKeyX = GlobalKey<FormState>();
+  final formKeyY = GlobalKey<FormState>();
+  final formKeyZ = GlobalKey<FormState>();
+  TextEditingController xTextController = TextEditingController();
+  TextEditingController yTextController = TextEditingController();
+  TextEditingController zTextController = TextEditingController();
+  final _dolbyioCommsSdkFlutterPlugin = DolbyioCommsSdk.instance;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(mainAxisSize: MainAxisSize.min, children: [
+      Row(
+        mainAxisAlignment: MainAxisAlignment.spaceAround,
+        children: [
+          const Text("x:"),
+          SizedBox(
+            width: 100,
+            child: Form(
+              key: formKeyX,
+              autovalidateMode: AutovalidateMode.disabled,
+              child: TextFormField(
+                  controller: xTextController..text = '0.0',
+                  keyboardType: TextInputType.number,
+                  inputFormatters: [
+                    FilteringTextInputFormatter.allow(RegExp(r'(^\d*\.?\d*)')),
+                  ]),
+            ),
+          )
+        ],
+      ),
```

Comment:
Could this be a widget? Or do we have a widget like this?

Reply:
Yes, we have but in Aga's PR. After merge Aga's code I can refactor this.